internal/services/apimanagement/api_management_api_diagnostic_resource.go

```diff
@@ -123,6 +124,16 @@ func resourceApiManagementApiDiagnostic() *pluginsdk.Resource {
 				}, false),
 			},
 		},
+
+		CustomizeDiff: func(ctx context.Context, d *pluginsdk.ResourceDiff, i interface{}) error {
+			if _, n := d.GetChange("operation_name_format"); n != "" {
+				if d.Get("identifier") != "applicationinsights" {
+					return fmt.Errorf("operation_name_format cannot be set when identifier is not applicationinsights")
```

Comment:
We usually put property names and values in error messages between back ticks 
```suggestion
					return fmt.Errorf("`operation_name_format` cannot be set when `identifier` is not `applicationinsights`")
```